File: example/handler.js
// Create an instance of Scala.js class
const handler = require("./example-fastopt.js").Handler();

exports.foo = function(event, context, callback) {
    const param = event.param;
    // Call Scala.js method
    const result = handler.foo();
    callback(null, result);
}
exports.bar = function(event, context, callback) {
    const param = event.param;
    // Call Scala.js method
    const result = handler.bar();
    callback(null, result);
}

File: example/src/main/scala/Handler.scala
package jkugiya.example

import scala.scalajs.js.annotation.{ JSExport, JSExportTopLevel }

@JSExportTopLevel("Handler")
class Handler {

  @JSExport
  def foo(): Unit =
    println("Foo!")

  @JSExport
  def bar(): Unit =
    println("Bar!")

}